### ISSUE TITLE
Artifacts/test logs aren't sharable

### ISSUE THREAD

**Issue by Kernald** [buildbuddy-io/buildbuddy#4]
Currently, clicking on an artifact or test log gives the path to the file on the local machine that ran the test - which doesn't help when the invocation ran on a different machine (e.g. coworker or CI).

As far as I understand, having those files reachable would require to either implement a CAS storage (maybe outside of the scope of this project?) or plug to an existing one (e.g. https://github.com/buildbarn/bb-storage).

**Comment by siggisim** [comment]
Hey @Kernald - this is high on our list! Sharing build artifacts is a high value use case.

Integrating a CAS is something we're exploring, but that likely wont be the only solution because many teams already upload their artifacts to another system like [JFrog Artifactory](https://jfrog.com/artifactory/) or another build cache - like BuildBarn.

I haven't done much digging here, but I'd love to find a way to integrate those urls into the BuildBuddy UI. An idea that comes to mind is some sort of URL pattern that you could specify in the config.yaml that would map local urls to remote urls. 

This would only work if the remote URLs are straightforward enough that they could easily be mapped (i.e. file://some/local/path/foo/bar/artifact.bin => https://cache.server.com/foo/bar/artifact.bin). The idea falls apart if the artifacts are stored at more involved urls (i.e. https://cache.server.com/run2736/foo/bar/artifact.bin) or randomized urls (https://cache.server.com/4j4nkh-f98jdfo-sdf98sj).

There are 3 interesting flags in the Bazel command line reference:
* [--build_event_json_file_path_conversion](https://docs.bazel.build/versions/master/command-line-reference.html#flag--build_event_binary_file_path_conversion)
* [--build_event_text_file_path_conversion](https://docs.bazel.build/versions/master/command-line-reference.html#flag--build_event_text_file_path_conversion)
* [--build_event_binary_file_path_conversion](https://docs.bazel.build/versions/master/command-line-reference.html#flag--build_event_binary_file_path_conversion)

The docs read:
```
Convert paths in the binary file representation of the build event protocol to 
more globally valid URIs whenever possible; if disabled, the file:// uri scheme 
will always be used
```

These flags seem to only affect the json, text, and binary build event output files respectively. As far as I can tell they don't seem to affect the build event protocol. I'm also not sure how this conversion is being handled, but it seems like a promising route to explore. 

Open to suggestions here!

**Comment by kzadorozhny** [comment]
There is an experimental flag to redirect log file to the remote storage:
https://docs.bazel.build/versions/master/command-line-reference.html#flag--experimental_stream_log_file_uploads

**Comment by siggisim** [comment]
This should be fixed in v1.0 - build artifacts are now clickable if you use BuildBuddy's built-in remote cache or a third party GRPC-enabled remote cache like bazel-remote. Test logs are also now viewable inline when a cache enabled!
